### PR TITLE
HackStudio: Update window close button on document change

### LINKED ISSUE #13285: Hack Studio does not honor  modified close button
Visual comparison of Hack Studio and Text Editor and of 'Close button (modified) '

image

image

I'm not sure if Hack Studio already has this feature previously, if not then disregard and close this issue with my apologies.

### PR DESCRIPTION
**Pull request by timre13**
Fixes #13285.

### CODE REVIEW

File: Userland/DevTools/HackStudio/HackStudioWidget.cpp [hunk 1]
```diff
@@ -1588,6 +1588,7 @@ void HackStudioWidget::update_tree_view()
 void HackStudioWidget::update_window_title()
 {
     window()->set_title(String::formatted("{} - {} - Hack Studio", m_current_editor_wrapper->filename_title(), m_project->name()));
+    window()->set_modified(any_document_is_dirty());
```
Comment (alimpfard) [inline Userland/DevTools/HackStudio/HackStudioWidget.cpp:1591]: This doesn't seem to be called reliably:
- Open a file and write into it, note that the window is not marked as modified
- Do that, switch to another document, switch back (note that it is marked as modified now), click the close button and select 'No'; note that the window is closed anyway.

Comment (timre13) [reply Userland/DevTools/HackStudio/HackStudioWidget.cpp:1591]: The `(*)` mark in the title is also not shown immediately (so we have an another bug). It seems that this function is called when opening a file. There should be a way to call it when modifying the file. Should I add some kind of callback which calls this?

Comment (timre13) [reply Userland/DevTools/HackStudio/HackStudioWidget.cpp:1591]: It think it works. How do I update the commit? I'm still new to this.

**Comment by timre13** [comment]
It seems to work now. This also fixes that the `(*)` mark only appeared when reopening the file.

**Comment by bgianfo** [comment]
💜 💙 💚

**Pull request merged**